backend/api/communication/src/test/java/co/airy/core/api/communication/ConversationsNoteTest.java

```diff
@@ -0,0 +1,240 @@
+    static void afterAll() throws Exception {
+        kafkaTestHelper.afterAll();
+    }
+
+    @BeforeEach
+    void beforeEach() throws Exception {
+        webTestHelper.waitUntilHealthy();
+    }
+
+    @Test
+    void canAddDeleteNote() throws Exception {
+        final Channel channel = Channel.newBuilder()
+                .setConnectionState(ChannelConnectionState.CONNECTED)
+                .setId(UUID.randomUUID().toString())
+                .setSource("facebook")
+                .setSourceChannelId("ps-id")
+                .build();
+
+        kafkaTestHelper.produceRecord(new ProducerRecord<>(applicationCommunicationChannels.name(), channel.getId(), channel));
+        final String conversationId = UUID.randomUUID().toString();
+        kafkaTestHelper.produceRecords(TestConversation.generateRecords(conversationId, channel, 1));
+
+        retryOnException(() -> webTestHelper.post("/conversations.info",
+                "{\"conversation_id\":\"" + conversationId + "\"}")
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id", is(conversationId))), "Conversation was not created");
+
+        final String text = "This is a test note";
+
+        webTestHelper.post("/conversations.addNote",
+                        "{\"conversation_id\":\"" + conversationId + "\",\"text\":\"" + text + "\"}")
+                .andExpect(status().isNoContent());
+
+        final String[] content = {""};
+
+        retryOnException(
+                () -> {
+                    MvcResult res = webTestHelper.post("/conversations.info",
+                                    "{\"conversation_id\":\"" + conversationId + "\"}")
+                            .andExpect(status().isOk())
+                            .andExpect(status().isOk())
+                            .andExpect(jsonPath("$.id", is(conversationId)))
+                            .andExpect(jsonPath("$.metadata.notes", is(not(nullValue()))))
+                            .andReturn();
+                    content[0] = res.getResponse().getContentAsString();
+                },
+                "Note was not added");
+
+        String noteKey = "";
+        Pattern pattern = Pattern.compile("(\"notes\":\\{\"){1}([a-z0-9-]*)(\":\"){1}");
+        Matcher matcher = pattern.matcher(content[0]);
+        if (matcher.find()) {
+            noteKey = matcher.group(2);
+        }
+
+        String finalNoteKey = noteKey;
+
+        webTestHelper.post("/conversations.info",
+                        "{\"conversation_id\":\"" + conversationId + "\"}")
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id", is(conversationId)))
+                .andExpect(jsonPath(String.format("$.metadata.notes['%s']", finalNoteKey), is(text)));
+
+        webTestHelper.post("/conversations.deleteNote",
+                        "{\"conversation_id\":\"" + UUID.fromString(conversationId) + "\",\"note_id\":\"" + UUID.fromString(finalNoteKey) + "\"}")
```

Comment:
What's the use of `UUID.fromString` here if it gets casted back into a string by the concatenation?